java/ql/src/semmle/code/java/dataflow/NullGuards.qll

```diff
@@ -24,6 +24,33 @@ Expr enumConstEquality(Expr e, boolean polarity, EnumConstant c) {
   )
 }
 
+/** Gets an instanceof expression of `v` with type `type` */
+InstanceOfExpr instanceofExpr(SsaVariable v, Expr type) {
```

Comment:
It is better to move the `getType()` of the type expression into this predicate, as that allows a more direct join in the use of this predicate.